src/Datapoint.c

```diff
@@ -124,11 +124,12 @@ int dpoint_to_binary(ds_datapoint_t *dpoint, unsigned char *buf, int *maxsize)
 {
   int total_bytes = 0;
   int bufidx = 0;
-  uint16_t varlen = strlen(dpoint->varname);
+
+  uint16_t varlen = strlen(dpoint->varname) + 1;
```

Comment:
This aligns `varlen` with the interpretation of `strlen` + 1.  We can be consistent about that here in dserv, but is there client code out there that needs to match?